### PR TITLE
Handle manually paused containers

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
We now handle manually paused containers (via `runc pause`) better,
which means:

- Pods with paused containers can be stopped by Kubernetes, because we
  now automatically resume them before attempting to stop the container.
- We're now able to handle the paused state in CRI-O internally by going
  back to `CONTAINER_RUNNING` if a user invoked `runc resume`. We also
  set the creation and started date accordingly to not throw Kubernetes
  events complaining that the dates are not set.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
Unfortunately we cannot easily detect if a container is in paused state without updating the container status internally. Because of that we have to trigger an RPC which automatically updates the status, like `execsync`.
#### Does this PR introduce a user-facing change?

```release-note
Improved handling of manually paused containers.
```

### CODE REVIEW

File: server/container_status.go [hunk 1]
```diff
@@ -54,7 +54,12 @@ func (s *Server) ContainerStatus(ctx context.Context, req *types.ContainerStatus
 
 	// If we defaulted to exit code not set earlier then we attempt to
 	// get the exit code from the exit file again.
-	if cState.Status == oci.ContainerStateStopped && cState.ExitCode == nil {
+	if (cState.Status == oci.ContainerStateStopped && cState.ExitCode == nil) ||
+		// Manually paused containers should be covered as well.
+		// Kubernetes has no way if displaying that information (state will be
```
Comment (TomSweeneyRedHat) [inline server/container_status.go:59]:
```suggestion
		// Kubernetes has no way of displaying that information (state will be
```

File: internal/oci/runtime_oci.go [hunk 2]
```diff
@@ -712,6 +712,18 @@ func (r *runtimeOCI) StopContainer(ctx context.Context, c *Container, timeout in
 		}
 	}()
 
+	// Paused containers cannot be stopped, so we have to resume them and
+	// update the container status accordingly.
+	if c.state.Status == ContainerStatePaused {
+		logrus.Warnf("Container %s is paused, resuming before trying to stop", c.ID())
+		if err := r.UnpauseContainer(ctx, c); err != nil {
+			return errors.Wrap(err, "resume container")
```
Comment (TomSweeneyRedHat) [inline internal/oci/runtime_oci.go:720]: would it be better to have "failed to resume container"?  If so, similar comment for other error messages.

File: server/container_status.go (hunk 1 above)
Comment (saschagrunert) [reply server/container_status.go]: Fixed as suggested :+1:

File: internal/oci/runtime_oci.go (hunk 2 above)
Comment (saschagrunert) [reply internal/oci/runtime_oci.go:720]: Well' I'd prefer reducing the amount of word repetitions in error messages. Usually the return value (an `error` type) already indicates that something went wrong. Assuming that those errors are chained, the resulting message would look like:
`error in x: did y: on my way to z` compared to: `error in x: failed to do y: failed to z`

**Pull request closed without merging**